docs/docs/examples/vault.md

```diff
@@ -0,0 +1,264 @@
+
+### Deploy Vault
+
+Now that the Coordinator has a manifest set, which defines the Vault deployment as an allowed workload,
+we can deploy the application:
+
+```sh
+kubectl apply -f deployment/
+```
+:::note[Persistent workload secrets]
+
+During the initialization process of the workload pod, the Contrast Initializer
+sends an attestation report to the Coordinator and receives a workload secret
+derived from the Coordinator's secret seed and the `workloadSecretID` specified in the
+manifest, and writes it to a secure in-memory `volumeMount`.
+
+:::
+<!-- TODO(jmxnzo): revise this paragraph as soon as we support filesystem volumes everywhere -->
+The Vault deployment is defined as a StatefulSet using the OpenBao Vault image, with a mounted block device for persistent storage.
+A Contrast Initializer, running as an init container, uses the workload secret located at `/contrast/secrets/workload-secret-seed` to
+generate an encryption key and initialize the block device as a LUKS-encrypted partition.
+Before the Vault container starts, the Initializer opens the LUKS device using the generated key.
+This unlocked device is then mounted by the Vault container and used as the backend storage volume.
+For the Vault application, this process is entirely transparent, and the device behaves like a standard volume mount.
+It’s important to clarify that the LUKS encryption of the block device is primarily a convenience feature, enabling persistent storage
+at the filesystem level on confidential virtual machines.
+The primary and security-relevant encryption mechanism remains Vault’s own sealing process, which provides cryptographic protection of
+secrets—even in the event that the underlying storage is compromised.
+
+Because the `workload-secret-seed` is derived from the associated `workloadSecretID`,
+any change to the `workloadSecretID` after the block device has been initialized will result in deriving an invalid encryption key,
+making the mounted block device undecryptable.
+
+:::note[Inter-deployment communication]
+
+The Contrast Coordinator issues mesh certificates after successfully validating workloads.
+These certificates can be used for secure inter-deployment communication.
+The Initializer sends an attestation report to the Coordinator,
+retrieves a service mesh certificate bound to it's provided public key, containing the certificate chain, as well as the current mesh CA cert.
+The Initializer then writes them to a `volumeMount`, allowing to build up the secure mTLS connections based on the service mesh.
+The received service mesh certificate also holds the certificate extension of the `workloadSecretID`,
+which is used to allow the authorization to a certain encryption key on the transit engine API.
+
+:::
+
+
+The Vault's TCP listener is configured to accept connections only from mesh certificates issued under the current state of the service mesh CA,
+effectively restricting communication to attested Contrast deployments.
```

Comment:
This should reflect that the CA trusted by Vault is the one which signed its (Vault's) own certificate, not the current one.